From b486e9f1e258d514ad6f6fd8d2be1f449e900516 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 10 Jan 2023 16:28:11 +0200
Subject: [PATCH] [SW-118475] preboot: add start_image() declaration to header

preboot_start_image() is imlemented in many source files but not declared
in any header. It compiled so far thanks to local weak implementation
but it should be declared in a header available to all users.

Change-Id: Id76644efc02b2144492619a7a12ae220fd1b7e87
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/include/preboot.h b/include/preboot.h
index 47f5ac2..ec0c517 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -226,6 +226,7 @@
 int preboot_verify_img_sig(const void *fit, size_t img_size,
 			   enum fit_sec_failure *res, u32 *key_idx);
 size_t preboot_get_img_sec_desc_size(void);
+void preboot_start_image(u64 startaddr, bool valid_arc_img);
 
 u32 preboot_error_read_and_clr(u32 *major, u32 *minor);
 bool preboot_error_inform(u32 major, u32 minor);
